arxiv/base/urls/links.py

```diff
@@ -302,11 +301,33 @@ def _compose_list_of_funcs(fv: List[Callable_Linker]) -> Callable_Linker:
     return reduce(lambda f, g: lambda x: f(g(x)), fv[1:], fv[0])
 
 
+def _deferred_thread_local_linker_of_kind(kind: str) -> Callable_Linker:
+    # Bleach linkifiers are not thread safe. This puts them on Flask.g.
+    #
+    # This will return a Callable that creates the bleach linkifier
+    # only when called with the string to linkify. This is so the
+    # bleach linkifier can be put into a Flask.g of each
+    # individual WSGI thread.
+
+    def deferred(instr: str) -> str:
+        # This must not be called while the app is starting up, only
+        # when the linkiers are being used in requests. So the
+        # function is created/cached at call time.
+        return g.get('linkers',{}).get(kind,_get_linker_of_kind(kind))(instr)
```

Comment:
This looks good. Not seeing where you set ``g.linkers`` though? Without that it looks like this will just create a new ``bleach.linkifier.Linker`` each time.

Reply:
Oh ya.